src/pip/_internal/build_env.py

```diff
@@ -188,6 +188,12 @@ def install(
                 )
             )
 
+        if finder.release_control is not None:
+            # Use ordered args to preserve the user's original command-line order
+            # This is important because later flags can override earlier ones
+            for attr_name, value in finder.release_control.get_ordered_args():
+                args.extend(("--" + attr_name.replace("_", "-"), value))
+
```

Comment:
I can't wait until we eliminate the subprocess build env installer so we can just let this configuration be inherited through the finder for free.